src/arch/svm/SpokeUtils.ts

```diff
@@ -67,17 +69,72 @@ export function getDepositIdAtBlock(_contract: unknown, _blockTag: number): Prom
   throw new Error("getDepositIdAtBlock: not implemented");
 }
 
-export function findDepositBlock(
-  _spokePool: unknown,
+/**
+ * Finds deposit events within a 2-day window ending at the specified timestamp.
+ *
+ * @remarks
+ * This implementation uses a time-limited search approach because Solana PDA state has
+ * limitations that prevent directly referencing old deposit IDs. Unlike EVM chains where
+ * we might use binary search across the entire chain history, in Solana we must query within
+ * a constrained slot range.
+ *
+ * The search window is calculated by:
+ * 1. Finding the slot at the specified timestamp (or current time if no timestamp is provided)
+ * 2. Looking back 2 days worth of slots from that point
+ *
+ * We use a 2-day window because:
+ * 1. Most valid deposits that need to be processed will be recent
+ * 2. This covers multiple bundle submission periods
+ * 3. It balances performance with practical deposit age
+ *
+ * @important
+ * This function may return `undefined` for valid deposit IDs that are older than the search
+ * window (approximately 2 days before the specified timestamp). This is an acceptable limitation
+ * as deposits this old are typically not relevant to current operations.
+ *
+ * @param eventClient - SvmCpiEventsClient instance
+ * @param depositId - The deposit ID to search for
+ * @param timestamp - The timestamp to search up to (defaults to current time). The search will look
+ *                   for deposits between (timestamp - 2 days) and timestamp. Time must be in seconds.
+ * @returns The deposit if found within the time window, undefined otherwise
+ */
+export async function findDeposit(
+  eventClient: SvmCpiEventsClient,
   depositId: BigNumber,
-  _lowBlock: number,
-  _highBlock?: number
-): Promise<number | undefined> {
+  timestamp: number = getCurrentTime()
```

Comment:
ooc, why do we set the bounds based on timestamp rather than ~block~ slot number?